docs/root/configuration/operations/tools/router_check.rst

```diff
@@ -183,3 +183,30 @@ fields that could be tested.
   > bazel-bin/test/tools/router_check/router_check_tool --config-path ... --test-path ... --useproto --f 7 --covall
   Current route coverage: 6.2948%
   Failed to meet coverage requirement: 7%
+
+When the fail under flag is set an optional flag (`-i` or `--increase-cov`) can
```

Comment:
I'm a little confused on what this does. How does the tool know what the current coverage is?

Reply:
Maybe I should revisit the docs. The current coverage is measured by checking how many tests we have versus how many routes we have. At the moment we can have the coverage check fail if the calculated coverage rate is below a specified number `x`%. This flag is to enforce that the specified number matches the calculated coverage to prevent a scenario outlined below.

1. Initial router_check is run and coverage is reported at 30%
2. `router_check --fail-under 30` is now used to enforce coverage doesn't drop below 30%
3. someone adds new tests which pass tests because the coverage is increased. However, they can forget to increase the fail-under percentage.
4. another person adds new routes without unit tests but tests pass because the coverage is still greater than 30%

Hopefully that clarifies things :)

Reply:
It actually doesn't. I don't understand how the tool can know that coverage was increased without someone telling it or it having access to the pre-PR state? Are we assuming the infrastructure runs things twice?

Reply:
This doesn't enforce that you increase the coverage of the tests. It's use is to enforce that you increase the value of the fail under argument, if one is present.

Reply:
Sorry for being dense but I still don't understand. How does one know to increase the fail under argument? The idea is that it fails and then you update it to the new value? If that's the case can you potentially redo the docs to make it more clear?

Reply:
@mattklein123 I attempted to make the intent of this flag clearer. Please take a look and let me know what you think.